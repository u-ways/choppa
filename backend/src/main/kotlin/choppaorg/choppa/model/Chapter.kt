package choppaorg.choppa.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import org.hibernate.annotations.GenericGenerator
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "chapter")
data class Chapter @JsonCreator constructor(
        @Id
        @Column(name = "chapter_id", columnDefinition = "uuid")
        @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
        @JsonProperty("id")
        val id: UUID,

        @Column(name = "name", columnDefinition = "VARCHAR(100)", unique = true, nullable = false)
        @JsonProperty("name")
        val name: String,

        @OneToMany(mappedBy = "chapter")
        @JsonIgnore
        val member: List<Member>
) {
        override fun toString() = "Chapter(id=$id, name=$name)"
}