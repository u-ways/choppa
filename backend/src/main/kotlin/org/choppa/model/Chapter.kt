package org.choppa.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import org.hibernate.annotations.GenericGenerator
import java.util.UUID
import java.util.UUID.randomUUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType.LAZY
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "chapter")
data class Chapter @JsonCreator constructor(
    @Id
    @Column(name = "chapter_id", columnDefinition = "uuid")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("id")
    val id: UUID = randomUUID(),

    @Column(name = "name", columnDefinition = "VARCHAR(100)", unique = true, nullable = false)
    @JsonProperty("name")
    val name: String,

    @OneToMany(mappedBy = "chapter", fetch = LAZY)
    @JsonIgnore
    val member: List<Member> = emptyList()
) {
    override fun toString() = "Chapter(id=$id, name=$name)"
}
