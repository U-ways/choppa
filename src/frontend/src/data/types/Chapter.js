export default class Chapter {
  constructor(id, name, colour) {
    this._id = id;
    this._name = name;
    this._colour = colour;
  }

  get id() {
    return this._id;
  }

  get name() {
    return this._name;
  }

  set name(newName) {
    this._name = newName;
  }

  get colour() {
    return this._colour;
  }

  set colour(newColour) {
    this._colour = newColour;
  }
}
