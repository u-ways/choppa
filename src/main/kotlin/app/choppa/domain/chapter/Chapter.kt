package app.choppa.domain.chapter

import app.choppa.domain.base.BaseModel
import app.choppa.domain.member.Member
import app.choppa.utils.Color.Companion.GREY
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.hibernate.annotations.GenericGenerator
import java.util.UUID
import java.util.UUID.fromString
import java.util.UUID.randomUUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "chapter")
@JsonSerialize(using = ChapterSerializer::class)
@JsonDeserialize(using = ChapterDeserializer::class)
data class Chapter(
    @Id
    @Column(name = "chapter_id", columnDefinition = "uuid")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    override val id: UUID = randomUUID(),

    @Column(name = "name", columnDefinition = "VARCHAR(100)", unique = true, nullable = false)
    val name: String = "CH-$id".substring(0, 15),

    @Column(name = "color", columnDefinition = "INTEGER")
    val color: Int = GREY,

    @OneToMany(mappedBy = "chapter")
    val members: MutableList<Member> = mutableListOf(),
) : BaseModel {
    override fun toString() = "Chapter(id=$id, name=$name)"

    override fun hashCode(): Int = id.hashCode()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Chapter
        if (id != other.id) return false
        return true
    }

    companion object {
        val UNASSIGNED_ROLE = Chapter(fromString("00000000-0000-0000-0000-000000000000"), "Unassigned")
    }
}
