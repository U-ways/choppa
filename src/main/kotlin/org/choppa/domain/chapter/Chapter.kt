package org.choppa.domain.chapter

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.choppa.domain.base.BaseModel
import org.choppa.domain.member.Member
import org.choppa.utils.Color.Companion.GREY
import org.hibernate.annotations.GenericGenerator
import java.util.UUID
import java.util.UUID.randomUUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "chapter")
@JsonSerialize(using = ChapterSerializer::class)
@JsonDeserialize(using = ChapterDeserializer::class)
data class Chapter @JsonCreator constructor(
    @Id
    @Column(name = "chapter_id", columnDefinition = "uuid")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("id")
    override val id: UUID = randomUUID(),

    @Column(name = "name", columnDefinition = "VARCHAR(100)", unique = true, nullable = false)
    @JsonProperty("name")
    val name: String = "CH-$id".substring(0, 15),

    @Column(name = "color", columnDefinition = "INTEGER")
    @JsonProperty("color")
    val color: Int = GREY,

    @OneToMany(mappedBy = "chapter")
    @JsonIgnore
    val members: MutableList<Member> = mutableListOf(),
) : BaseModel {
    override fun toString() = "Chapter(id=$id, name=$name)"

    override fun hashCode(): Int = id.hashCode()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Chapter
        if (id != other.id) return false
        return true
    }

    companion object {
        val UNASSIGNED_ROLE = Chapter(UUID.fromString("00000000-0000-0000-0000-000000000000"), "Unassigned")
    }
}
