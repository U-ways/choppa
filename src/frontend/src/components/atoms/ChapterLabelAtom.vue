<template>
  <div class="label text-center" :style="cssVars">
    {{ chapter.name }}
  </div>
</template>

<script>
import Chapter from "@/data/types/Chapter";

export default {
  name: "ChapterLabelAtom",
  props: {
    chapter: Chapter,
  },
  computed: {
    cssVars() {
      return {
        "--bg-colour": this.chapter.colour,
      };
    },
  },
};
</script>

<style scoped>

.label {
  color: white;
  border: 3px solid black;
  border-radius: 5px 0 5px 0;
  padding: 5px;
  background-color: var(--bg-colour);
  width: 100%;
}
</style>
