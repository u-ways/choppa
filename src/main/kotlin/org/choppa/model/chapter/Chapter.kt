package org.choppa.model.chapter

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.choppa.model.Member
import org.hibernate.annotations.GenericGenerator
import java.util.UUID
import java.util.UUID.randomUUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "chapter")
@JsonSerialize(using = Serializer::class)
@JsonDeserialize(using = Deserializer::class)
data class Chapter @JsonCreator constructor(
    @Id
    @Column(name = "chapter_id", columnDefinition = "uuid")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("id")
    val id: UUID = randomUUID(),

    @Column(name = "name", columnDefinition = "VARCHAR(100)", unique = true, nullable = false)
    @JsonProperty("name")
    val name: String,

    @OneToMany(mappedBy = "chapter")
    @JsonIgnore
    val members: List<Member> = emptyList(),
) {
    override fun toString() = "Chapter(id=$id, name=$name)"
}
