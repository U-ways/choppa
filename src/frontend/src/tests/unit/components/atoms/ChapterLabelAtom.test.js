import { describe, expect, it, shallowMount } from "@/tests/unit/test-imports";
import ChapterLabelAtom from "@/components/atoms/ChapterLabelAtom";
import Chapter from "@/data/types/Chapter";

describe("Chapter Label Atom test", () => {
  it("Renders correctly", async () => {
    const wrapper = shallowMount(ChapterLabelAtom, {
      propsData: {
        chapter: new Chapter(1, "Chapter Test", "#FF00FF"),
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
