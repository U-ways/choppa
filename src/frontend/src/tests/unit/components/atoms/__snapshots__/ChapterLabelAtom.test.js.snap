// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter Label Atom test Renders correctly 1`] = `
"<div class=\\"label text-center\\" style=\\"--bg-colour: #FF00FF;\\">
  Chapter Test
</div>"
`;
